torch/fx/experimental/symbolic_shapes.py

```diff
@@ -195,6 +196,24 @@ def guard_scalar(a):
 
 # inclusive both ways
 def constrain_range(a, *, min: Optional[int], max: Optional[int] = None):
+    """
+    Constrain range takes a symbol, and records a valid value range for it in the graph.
+    That value range is the narrowest intersection of the current value range for the symbol.
+
+    Ex:
+    constrain_range(x, min=3, max=10)
+    constrain_range(x, min=4, max=12)
+
+    Would have a range of (4, 10)
```

Comment:
nit: conventionally, ( means exclusive, and [ means inclusive. So [4, 10] indicates an inclusive-inclusive range